pop-api/examples/dao/src/lib.rs

```diff
@@ -0,0 +1,364 @@
+				return Err(Error::ProposalAlreadyExecuted);
+			}
+
+			if proposal.yes_votes > proposal.no_votes {
+				let contract = self.env().account_id();
+
+				// Execute the proposal
+				let treasury_balance = api::balance_of(self.token_id, contract).unwrap_or_default();
+				if treasury_balance < proposal.amount {
+					// Not enough tokens in treasury to distribute
+					return Err(Error::TryAgainLater);
+				}
+
+				api::transfer(self.token_id, proposal.beneficiary, proposal.amount)
+					.map_err(Psp22Error::from)?;
+				self.env().emit_event(Transfer {
+					from: Some(contract),
+					to: Some(proposal.beneficiary),
+					value: proposal.amount,
+				});
+				self.env().emit_event(Approval {
+					owner: contract,
+					spender: contract,
+					value: proposal.amount,
+				});
+
+				proposal.executed = true;
+
+				self.proposals.remove(proposal_id);
+				self.proposals.insert(proposal_id, &proposal);
+				Ok(())
+			} else {
+				Err(Error::ProposalRejected)
+			}
+		}
+
+		/// Allows a user to become a member of the Dao
+		/// by transferring some tokens to the DAO's treasury.
+		/// The amount of tokens transferred will be stored as the
+		/// voting power of this member.
+		///
+		/// # Parameters
+		/// - `amount` - Balance transferred to the Dao and representing
+		/// the voting power of the member.
+
+		#[ink(message)]
+		pub fn join(&mut self, amount: Balance) -> Result<(), Error> {
+			let caller = self.env().caller();
+			let contract = self.env().account_id();
+			api::transfer_from(self.token_id, caller, contract, amount)
+				.map_err(Psp22Error::from)?;
+			let member =
+				self.members.get(caller).unwrap_or(Member { voting_power: 0, last_vote: 0 });
+
+			let voting_power =
+				member.voting_power.checked_add(amount).ok_or(Error::ArithmeticOverflow)?;
+			self.members
+				.insert(caller, &Member { voting_power, last_vote: member.last_vote });
+
+			self.env().emit_event(Transfer {
+				from: Some(caller),
+				to: Some(contract),
+				value: amount,
+			});
+
+			Ok(())
+		}
+
+		#[ink(message)]
+		pub fn get_member(&mut self, account: AccountId) -> Member {
+			self.members.get(account).unwrap_or(Member { voting_power: 0, last_vote: 0 })
+		}
+
+		#[ink(message)]
+		pub fn get_block_number(&mut self) -> Option<BlockNumber> {
+			Some(self.env().block_number())
+		}
+
+		#[ink(message)]
+		pub fn positive_votes(&mut self, proposal_id: u32) -> Option<Balance> {
+			let proposal = &self.proposals.get(proposal_id).unwrap();
+			Some(proposal.yes_votes)
+		}
+
+		#[ink(message)]
+		pub fn negative_votes(&mut self, proposal_id: u32) -> Option<Balance> {
+			let proposal = &self.proposals.get(proposal_id).unwrap();
+			Some(proposal.no_votes)
+		}
+	}
+
+	#[derive(Debug, PartialEq, Eq)]
+	#[ink::scale_derive(Encode, Decode, TypeInfo)]
+	pub enum Error {
+		ArithmeticOverflow,
+		ProposalNotFound,
+		VotingPeriodEnded,
+		NotAMember,
+		AlreadyVoted,
+		VotingPeriodNotEnded,
+		ProposalAlreadyExecuted,
```

Comment:
```suggestion
		ProposalExecuted,
```
to keep it consistent with `ProposalRejected`